src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -167,12 +167,13 @@ internal Batch<Metric> Collect()
                     // Record all observable instruments
                     this.listener.RecordObservableInstruments();
                     var indexSnapShot = Math.Min(this.metricIndex, MaxMetrics - 1);
-                    for (int i = 0; i < indexSnapShot + 1; i++)
+                    var target = indexSnapShot + 1;
+                    for (int i = 0; i < target; i++)
                     {
                         this.metrics[i].SnapShot();
                     }
 
-                    return new Batch<Metric>(this.metrics, indexSnapShot + 1);
+                    return (target > 0) ? new Batch<Metric>(this.metrics, indexSnapShot + 1) : default;
```

Comment:
nit: Looks like you could use target in here too (instead of `indexSnapShot + 1`): `new Batch<Metric>(this.metrics, target)`